### PR TITLE
fix: Ensure Dev*,Microcom and Build Task are builds before Controls

### PR DESCRIPTION
**Pull request by workgroupengineering**
## What does the pull request do?

Ensure Dev*,Microcom and Build Task are builds before Controls when build solution in IDE

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0026200-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: Avalonia.sln [hunk 1]
```diff
@@ -12,6 +12,11 @@ EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Avalonia.Direct2D1", "src\Windows\Avalonia.Direct2D1\Avalonia.Direct2D1.csproj", "{3E908F67-5543-4879-A1DC-08EACE79B3CD}"
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Avalonia.Controls", "src\Avalonia.Controls\Avalonia.Controls.csproj", "{D2221C82-4A25-4583-9B43-D791E3F6820C}"
+	ProjectSection(ProjectDependencies) = postProject
```
Comment (kekekeks) [inline Avalonia.sln:15]: I think it should be built before Avalonia.Base.

It's weird how .NET 7 SDK somehow skips explicit ProjectReference nodes

Comment (fforjan) [reply Avalonia.sln:15]: @kekekeks  do you mind to expands ont the ProjectReference node ? when looking at that code change, I was wondering why you didn't use project reference but use solution project dependencies as by experience we do not use this on our side as tools which only look at projects will not be able to use this

Comment (kekekeks) [reply Avalonia.sln:15]: We _are_ using ProjectReference nodes and it used to work perfectly fine with .NET 6 SDK.

https://github.com/AvaloniaUI/Avalonia/blob/master/build/SourceGenerators.props#L3
https://github.com/AvaloniaUI/Avalonia/blob/master/build/DevAnalyzers.props

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0026236-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: Avalonia.sln (hunk 1 above)
Comment (grokys) [reply Avalonia.sln]: Not really, I encountered a problem several times that I had to explicitly rebuild e.g. Avalonia.Build.Tasks after switching branch before we move to .net 7.

**Review by maxkatz6 — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0026251-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Pull request merged**